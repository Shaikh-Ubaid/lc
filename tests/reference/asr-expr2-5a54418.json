{
    "basename": "asr-expr2-5a54418",
    "cmd": "lc --show-asr --no-color {infile} -o {outfile} -extra-arg=\"\"",
    "infile": "tests/expr2.c",
    "infile_hash": "11cb114715d67a4763454167abc900089b2119ecf18a980d18a30322",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr2-5a54418.stdout",
    "stdout_hash": "cdcd7c21f44822aca31bc8f5901b18bab75a9c340ad0c17b70ec12fa",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}