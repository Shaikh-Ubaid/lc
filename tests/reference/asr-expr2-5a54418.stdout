(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(=
                        (Var 2 x)
                        (IntegerBinOp
                            (IntegerBinOp
                                (IntegerConstant 2 (Integer 4))
                                Add
                                (IntegerConstant 3 (Integer 4))
                                (Integer 4)
                                ()
                            )
                            Mul
                            (IntegerConstant 5 (Integer 4))
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 2 x)]
                        ()
                        ()
                    )
                    (=
                        (Var 2 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            exit_code:
                                (Variable
                                    3
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(=
                        (Var 3 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
